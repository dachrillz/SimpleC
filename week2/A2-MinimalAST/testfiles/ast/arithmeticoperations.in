int a(){
    x = 1+2;
    y = x-2;
    q = 5*a;
    c = 5-1;
    return;
}
