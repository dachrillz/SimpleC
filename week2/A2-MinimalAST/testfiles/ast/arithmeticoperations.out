Program
  List
    Function ID="a"
      null
      List
        IdUse ID="x"
          Plus
            Var ID="1"
            Var ID="2"
        IdUse ID="y"
          Minus
            Var ID="x"
            Var ID="2"
        IdUse ID="q"
          Mul
            Var ID="5"
            Var ID="a"
        IdUse ID="c"
          Minus
            Var ID="5"
            Var ID="1"
        IdUse ID="d"
          Mod
            Var ID="5"
            Var ID="5"
        Return
          null
