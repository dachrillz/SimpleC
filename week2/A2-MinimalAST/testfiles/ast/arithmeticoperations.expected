Program
  List
    Function ID="a"
      List
        ExpressionStatement
          IdUse ID="x"
            Plus
              Numeral NUMERAL="1"
              Numeral NUMERAL="2"
        ExpressionStatement
          IdUse ID="y"
            Minus
              Var ID="x"
              Numeral NUMERAL="2"
        ExpressionStatement
          IdUse ID="q"
            Mul
              Numeral NUMERAL="5"
              Var ID="a"
        ExpressionStatement
          IdUse ID="c"
            Minus
              Numeral NUMERAL="5"
              Numeral NUMERAL="1"
        Return
